micro_sam/sam_annotator/_state.py

```diff
@@ -75,7 +75,8 @@ def initialize_predictor(
         assert ndim in (2, 3)
         # Initialize the model if necessary.
         if predictor is None:
-            self.predictor = util.get_sam_model(
+            get_model = util.get_sam_model if checkpoint_path is None else util.get_custom_sam_model
```

Comment:
I was thinking this line would be the one taking care of initializing the models flexlibly, but I see your point. It makes to stay consistent.